rest-dashboard.go

```diff
@@ -308,6 +336,94 @@ func (r *Client) DeleteDashboard(ctx context.Context, slug string) (StatusMessag
 	return reply, err
 }
 
+type (
+	// SearchParam is a type for specifying SearchWithParams params.
+	SearchParam func(*url.Values)
+	// SearchType is a type accepted by WithSearchType func.
+	SearchType string
```

Comment:
Maybe let's not export this type so that the users would be forced to choose from constants? i.e. it will prevent someone from doing `sdk.SearchType("whateversearchtype")`

Reply:
Should we not export this type, we will have some problems:
- user will never know about it since type definition would not be exported by godoc
- unexported types are a pain to use
- `go vet` won't like it: `searchType not exported by package sdk`
- user might still shoot his own leg off with something like this:
```
s := sdk.SearchDashboard
s = "searchSomething"
```
https://play.golang.org/p/l6LEzgBw_r7

Since we don't yet have any sort of string sets / enums in the language, I know of no method to achieve compile-time safety here.


As a point of reference: standard library seems not to count this as a problem. For example `net/http`'s `Request` just defines `Method` as as string:
https://golang.org/pkg/net/http/#Request

Do you think we should adopt the same approach here?